app/src/app-logic.js

```diff
@@ -1,16 +1,22 @@
-import React, { useCallback, useMemo, useState } from 'react'
-import { useAragonApi, useConnectedAccount } from '@aragon/api-react'
+import React, { useCallback } from 'react'
+import {
+  AragonApi,
+  useApi,
+  useAppState,
+  useCurrentApp,
+} from '@aragon/api-react'
+import appStateReducer from './app-state-reducer'
 import usePanelState from './hooks/usePanelState'
-import useUnwrapPanelState from './hooks/usePanelState'
 
 function noop() {}
 
 // Unwrap tokens action
 export function useUnwrapTokensAction(onDone = noop) {
-  const { api } = useAragonApi()
+  const api = useApi()
   return useCallback(
-    async amount => {
-      await api.unlock(amount).toPromise()
```

Comment:
We generally don't use the `await` here, so that the `onDone()` action isn't dependent on the promise passing.

If this intent failed, we would never call `onDone()` and never close the panel. We should instead catch errors in the panel before the user submits (see #8).